lib/src/backend.rs

```diff
@@ -147,6 +147,7 @@ content_hash! {
         pub description: String,
         pub author: Signature,
         pub committer: Signature,
+        pub sig: Option<Vec<u8>>,
```

Comment:
> Just extract the signature to be later (somehow) shown by the templater, for
example, and used otherwise.

To be able to verify it, we'll need the backend's help. That can come later, but I thought I'd mention it now in case it affects the design. I had not thought about it earlier. Any thoughts on how we would do that? Would the backend have a `verify_commit_signature(&self, commit: &Commit, valid: &dyn Fn(data: &[u8], signature: &[u8]) -> bool)`? Or is it better to have a `get_bytes_to_sign(&self, commit: &Commit) -> Vec<u8>` which could then be used for both the signing and verification paths? When writing a signed commit, we could then populate the commit, including a field for the signature message, and then call `get_bytes_to_sign()` and sign those bytes and update the commit object with them before calling `write_commit()`. I don't know if such an interface is generic enough.

Reply:
Yes I was thinking about having something like `get_bytes_to_sign`, for example git just gives you the `commit_create_buffer` which creates exactly the commit text that has to be signed/verified.